### PR TITLE
Field error property paths

### LINKED ISSUE #49: Suggestions on List field validation?
Example:

```java
@Service
class SomeService { 
   void doSomething(List<@Valid SomeItem> someItems) {}
}
```

Currently, `ApiFieldError` doesn't have a place to indicate which item within a list has the validation issue. Would it be worth adding an optional property of `ApiFieldError` like `index` to indicate the index of the problematic object in the List? I realize `index` may not help with `Map<String, @Valid SomeItem>`, for instance, or other structures. Currently we've had to reimplement `ConstraintViolationApiExceptionHandler`.

Thanks!

### PR DESCRIPTION
**Pull request by sdoxsee**
Closes #49

### CODE REVIEW

**Comment by wimdeblauwe** [comment]
Looks great, any idea why GitHub is showing all those old commits in this PR?

**Comment by sdoxsee** [comment]
I probably need to rebase better somewhere. My open source contribution skills a rusty :) Let me try again

**Comment by sdoxsee** [comment]
Force-pushed after rebasing on your `develop` branch. Single commit now. Thanks!

**Comment by sdoxsee** [comment]
Before merging though @wimdeblauwe , did you want to make `path` always appear or make its presence configurable?

Some other notes:
* I only added it for `fieldErrors` as `parameterErrors` would typically only be 1 deep?
* I sorted the fieldErrors by property path because their order seemed non-deterministic.
* I'm only handling `ConstraintViolation` in `BindApiExceptionHandler` so other sources of fieldErrors will lead to a null `path`

**Comment by wimdeblauwe** [comment]
@sdoxsee I did 2 extra commits:

1) Extra test with further nested objects to be sure that works fine.
2) There is now a property `addPathToError` which is `true` by default, but disables adding the `path` to the response when set to `false`, just in case somebody really does not like to have it there :)

**Comment by wimdeblauwe** [comment]
Let me know if you have any further remarks. I think this PR can be merged as is now.

**Comment by sdoxsee** [comment]
Meets my needs! Thanks for all your help and brainstorming. Cheers
